cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DistributedPreviewManager.java

```diff
@@ -157,22 +160,45 @@ public void run() {
             hConf.writeXml(writer);
           }
 
-          ResourceSpecification resourceSpec = ResourceSpecification.Builder.with()
+          Boolean artifactLocalizerEnabled = cConf.getBoolean(Constants.Preview.ARTIFACT_LOCALIZER_ENABLED);
+
+          ResourceSpecification runnerResourceSpec = ResourceSpecification.Builder.with()
             .setVirtualCores(cConf.getInt(Constants.Preview.CONTAINER_CORES))
             .setMemory(cConf.getInt(Constants.Preview.CONTAINER_MEMORY_MB), ResourceSpecification.SizeUnit.MEGA)
             .setInstances(cConf.getInt(Constants.Preview.CONTAINER_COUNT))
             .build();
 
-          LOG.info("Starting preview runners with {} instances", resourceSpec.getInstances());
+          Optional<ResourceSpecification> artifactLocalizerResourceSpec = Optional.empty();
+          if (artifactLocalizerEnabled) {
+            artifactLocalizerResourceSpec = Optional.of(
+              ResourceSpecification.Builder.with()
+                .setVirtualCores(cConf.getInt(Constants.ArtifactLocalizer.CONTAINER_CORES))
+                .setMemory(cConf.getInt(Constants.ArtifactLocalizer.CONTAINER_MEMORY_MB),
+                           ResourceSpecification.SizeUnit.MEGA)
+                .setInstances(cConf.getInt(Constants.TaskWorker.CONTAINER_COUNT))
+                .build());
+          }
+
+          LOG.info("Starting preview runners with {} instances", runnerResourceSpec.getInstances());
```

Comment:
nit: add whether sidecard container is also being used or not to the log.

Reply:
Done